web3.js/src/transaction.ts

```diff
@@ -395,13 +395,16 @@ export class Transaction {
 
     // Sort. Prioritizing first by signer, then by writable
     uniqueMetas.sort(function (x, y) {
-      const pubkeySorting = x.pubkey
-        .toBase58()
-        .localeCompare(y.pubkey.toBase58());
-      const checkSigner = x.isSigner === y.isSigner ? 0 : x.isSigner ? -1 : 1;
-      const checkWritable =
-        x.isWritable === y.isWritable ? pubkeySorting : x.isWritable ? -1 : 1;
```

Comment:
The CPU and memory that we consumed to produce base58 strings gets thrown out here if the signedness or writability of the accounts differ.